modules/extensions-support/src/main/java/org/mule/module/extension/internal/model/property/ConnectionHandlingTypeModelProperty.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.module.extension.internal.model.property;
+
+import org.mule.module.extension.internal.introspection.utils.ConnectionHandlingDescriptor;
+
+/**
+ * A immutable model property for {@link org.mule.extension.api.introspection.ConnectionProviderModel} which holds
+ * a immutable {@link ConnectionHandlingDescriptor} that can be accessed from {@link ConnectionHandlingTypeModelProperty#getDescriptor()}
+ * and contains the Connection Handling Type of the given Connection Provider
+ *
+ * @since 4.0
+ */
+public class ConnectionHandlingTypeModelProperty
```

Comment:
I don't see the need for this class + the descriptor. Move the descriptor contents into this